pkg/neonvm/controllers/vm_controller.go

```diff
@@ -553,18 +560,35 @@ func (r *VMReconciler) doReconcile(ctx context.Context, vm *vmv1.VirtualMachine)
 				return err
 			}
 
-			// get CPU details from QEMU
-			cpuSlotsPlugged, _, err := QmpGetCpus(QmpAddr(vm))
+			// get cgroups CPU details from runner pod
+			cgroupUsage, err := getRunnerCPULimits(ctx, vm)
 			if err != nil {
-				log.Error(err, "Failed to get CPU details from VirtualMachine", "VirtualMachine", vm.Name)
+				log.Error(err, "Failed to get CPU details from runner", "VirtualMachine", vm.Name)
 				return err
 			}
-			pluggedCPU := uint32(len(cpuSlotsPlugged))
+			var pluggedCPU uint32
 
-			// get cgroups CPU details from runner pod
-			cgroupUsage, err := getRunnerCgroup(ctx, vm)
-			if err != nil {
-				log.Error(err, "Failed to get CPU details from runner", "VirtualMachine", vm.Name)
+			if vm.Spec.CpuScalingMode == nil { // should not happen
+				err := fmt.Errorf("CPU scaling mode is not set")
+				log.Error(err, "Unknown CPU scaling mode", "VirtualMachine", vm.Name)
+				return err
+			}
+
+			switch *vm.Spec.CpuScalingMode {
+			case vmv1.CpuScalingModeSysfs:
+				log.Info("CPU usage check based on cgroups", "CpuScalingMode", *vm.Spec.CpuScalingMode)
```

Comment:
question: What's going on here? Saw "sysfs" + "based on cgroups" and was confused :sweat_smile:

Reply:
yeah, it is a consequence of the fact that all helper functions to call runner used to be called with Cgroups in it, like setRunnerGroups/getRunnerGroups, etc and eventually, it reached to the current state. Changed the log message.

Reply:
In fact, I don't think we really need those logs at all